From 0ef1a1b8bf92bffa15ddfac4030869693ecde404 Mon Sep 17 00:00:00 2001
From: saranya <saranya.gopal@intel.com>
Date: Tue, 15 May 2018 15:32:46 +0530
Subject: [PATCH 5/5] MtpPacket: Add finite timeout for bulk transfer

When timeout is set as 0 for bulk transfer,
kernel waits indefinitely resulting in hung_task
kernel panic in case of faulty responder devices.
This patch adds a finite timeout value for
bulk transfer to fix the kernel panic.

Change-Id: I8bda79e94a7c23ab319e49c72cbc89c80fc056d2
Tracked-On:
Signed-off-by: saranya <saranya.gopal@intel.com>
---
 media/mtp/MtpPacket.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/mtp/MtpPacket.cpp b/media/mtp/MtpPacket.cpp
index 3dd4248..f069a83 100644
--- a/media/mtp/MtpPacket.cpp
+++ b/media/mtp/MtpPacket.cpp
@@ -157,7 +157,7 @@ int MtpPacket::transfer(struct usb_request* request) {
                             request->endpoint,
                             request->buffer,
                             request->buffer_length,
-                            0);
+                            5000);
     request->actual_length = result;
     return result;
 }
-- 
1.9.1

